### PR TITLE
Add parcelable to viewpoint persistence

### PR DESCRIPTION
**Pull request by hud10837**
Adds a `Parcelable` implementation to `ViewpointPersistence`. This is the easiest way to resolve the problem where if the user intends to use `mutableStateOf(ViewpointPersistence...)` then they will have to use `rememberSaveable`. Rather than try to implement our own `remember` function or give the user a custom `Saver` or save functions, we can use `Parcelable` so that `rememberSaveable` automatically works without further implementation requirements.

### CODE REVIEW

**Comment by hud10837** [comment]
Draft until the feature branch is merged into `v.next`

File: toolkit/geo-compose/src/androidTest/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistenceTests.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        val fromParcelNone = ViewpointPersistence.None.CREATOR.createFromParcel(parcel)
+        val fromParcelByCenterAndScale = ViewpointPersistence.ByCenterAndScale.CREATOR.createFromParcel(parcel)
+        val fromParcelByBoundingGeometry = ViewpointPersistence.ByBoundingGeometry.CREATOR.createFromParcel(parcel)
+
+        // Assert that the read back instances are equal to the original ones
+        assertEquals(none, fromParcelNone)
+        assertEquals(byCenterAndScale, fromParcelByCenterAndScale)
+        assertEquals(byBoundingGeometry, fromParcelByBoundingGeometry)
+    }
+}
```
Comment (gunt0001) [inline toolkit/geo-compose/src/androidTest/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistenceTests.kt:35]: missing newline

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistence.kt [hunk 2]
```diff
@@ -17,23 +17,42 @@
...
 /**
  * Enum class representing the different types of viewpoint persistence on a composable [MapView].
  *
  * Viewpoint persistence determines how the viewpoint of a MapView is saved and restored across activity
  * or process recreation, for example, when the device is rotated or when the app is sent to the background
  * and then brought back to the foreground.
  *
+ * If the [ViewpointPersistence] is implemented with a [MutableState] type, it must be wrapped in [rememberSaveable],
+ * otherwise the state will not be saved and restored across activity or process recreation and the
+ * behavior of the [MapView] will be unpredictable.
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistence.kt:32]: Maybe add a sentence about the fact that ViewpointPersistence implements Parcelable.

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistence.kt [hunk 3]
```diff
@@ -17,23 +17,42 @@
...
-    public object None : ViewpointPersistence()
+    public object None : ViewpointPersistence() {
+        override fun describeContents(): Int = 0
+
+        override fun writeToParcel(dest: Parcel, flags: Int) {
+            // No state to write for this object
+        }
+
+        @JvmField
+        public val CREATOR: Parcelable.Creator<None> = object : Parcelable.Creator<None> {
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistence.kt:51]: Should it be a nullable type parameter?
```suggestion
        public val CREATOR: Parcelable.Creator<None?> = object : Parcelable.Creator<None?> {
```

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistence.kt [hunk 4]
```diff
@@ -17,23 +17,43 @@
...
  * Enum class representing the different types of viewpoint persistence on a composable [MapView].
  *
  * Viewpoint persistence determines how the viewpoint of a MapView is saved and restored across activity
  * or process recreation, for example, when the device is rotated or when the app is sent to the background
  * and then brought back to the foreground.
  *
+ * Note that a [MutableState] of [ViewpointPersistence] can not be used with [remember], because it will
+ * not be able to restore the state across process recreation. Instead, use [MutableState] of [Viewpoint]
+ * inside of [rememberSaveable] or within a [ViewModel]. Note that this class implements [Parcelable] so it
+ * can be used with [rememberSaveable] without any need for a custom [Saver].
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/ViewpointPersistence.kt:33]: Nice 👍

**Review by gunt0001 — APPROVED** [review]
@hud10837 - looks good 👍

**Review by changanxian — APPROVED** [review]
LGTM

**Comment by hud10837** [comment]
Thanks @changanxian @gunt0001

**Pull request merged**